qiskit/providers/ibmq/credentials/configrc.py

```diff
@@ -65,29 +66,35 @@ def read_credentials_from_qiskitrc(
     # Build the credentials dictionary.
     credentials_dict = OrderedDict()  # type: ignore[var-annotated]
     default_provider_hgp = None
+    credentials_count = 0
     for name in config_parser.sections():
-        single_credentials = dict(config_parser.items(name))
-
-        # Individually convert keys to their right types.
-        # TODO: consider generalizing, moving to json configuration or a more
-        # robust alternative.
-        if 'proxies' in single_credentials.keys():
-            single_credentials['proxies'] = literal_eval(
-                single_credentials['proxies'])
-        if 'verify' in single_credentials.keys():
-            single_credentials['verify'] = bool(  # type: ignore[assignment]
-                single_credentials['verify'])
-        if 'default_provider' in single_credentials.keys():
-            default_provider_hgp = HubGroupProject.from_stored_format(
-                single_credentials['default_provider'])
-
-            # Delete `default_provider`, since it's not used by the `Credentials` constructor.
-            del single_credentials['default_provider']
-
-        new_credentials = Credentials(**single_credentials)  # type: ignore[arg-type]
-
-        credentials_dict[new_credentials.unique_id()] = new_credentials
-
+        if name.startswith('ibmq'):
+            single_credentials = dict(config_parser.items(name))
+            credentials_count += 1
+            # Individually convert keys to their right types.
+            # TODO: consider generalizing, moving to json configuration or a more
+            # robust alternative.
+            if 'proxies' in single_credentials.keys():
+                single_credentials['proxies'] = literal_eval(
+                    single_credentials['proxies'])
+            if 'verify' in single_credentials.keys():
+                single_credentials['verify'] = bool(  # type: ignore[assignment]
+                    single_credentials['verify'])
+            if 'default_provider' in single_credentials.keys():
+                default_provider_hgp = HubGroupProject.from_stored_format(
+                    single_credentials['default_provider'])
+
+                # Delete `default_provider`, since it's not used by the `Credentials` constructor.
+                del single_credentials['default_provider']
+
+            new_credentials = Credentials(**single_credentials)  # type: ignore[arg-type]
+
+            credentials_dict[new_credentials.unique_id()] = new_credentials
+
+    if credentials_count > 1:
+        warnings.warn("Loading from multiple accounts is deprecated and "
+                      "will no longer be supported in future releases.",
+                      DeprecationWarning, stacklevel=4)
```

Comment:
You don't actually need to raise a warning as the existing code already [raises an error](https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/qiskit/providers/ibmq/ibmqfactory.py#L177). My comment from the other PR was about people using `ibmq_foo_bar` instead of just `ibmq` for as the section name. Nonetheless, this achieves your original goal of allowing credentials from different providers, as long as they don't start with `ibmq`. 

Can you add a release note (e.g. the one from the other PR)?

Reply:
Now I am puzzled.  I thought you were concerned because it was not deprecated, but it looks like it has been raising for 17 months.  So while not technically deprecated, anyone who uses it gets raised.  I can add a note and remove the warning.

Reply:
My concerned was about the _name_ of the section, not the number of sections in qiskitrc. The code in your previous PR ignored any section that's not named `ibmq`. And my argument was that the section could be named `ibmq_hub_group_project`, which is the format we used to have and never deprecated.